nixos/modules/services/web-apps/reposilite.nix

```diff
@@ -0,0 +1,253 @@
+            default = 30000;
+            example = 120000;
+            type = types.int;
+          };
+
+          bypassExternalCache = mkOption {
+            description = ''
+              Adds cache bypass headers to each request from /api/* scope served by this instance.
+              Helps to avoid various random issues caused by proxy provides (e.g. Cloudflare) and browsers.
+            '';
+            default = true;
+            type = types.bool;
+          };
+
+          cachedLogSize = mkOption {
+            description = "Amount of messages stored in cached logger.";
+            default = 32;
+            example = 64;
+            type = types.int;
+          };
+
+          defaultFrontend = mkOption {
+            description = "Enable default frontend with dashboard.";
+            default = true;
+            type = types.bool;
+          };
+
+          basePath = mkOption {
+            description = ''
+              Set custom base path for Reposilite instance.
+              It's not recommended to mount Reposilite under custom base path and you should always prioritize subdomain over this option.
+            '';
+            default = "/";
+            example = "/maven";
+            type = types.str;
+          };
+
+          debugEnabled = mkOption {
+            description = "Debug mode";
+            default = false;
+            type = types.bool;
+          };
+
+        };
+      };
+      description = "Configuration written to the config file for Reposilite.";
+      default = { };
+    };
+
+    tokens = mkOption {
+      description = ''
+        Tokens to use for authentication.
+        Passed to Reposilite via the `--token` option, in the format <name>:<value>.
+        It is recommended to add a least one token here, so that you can access the dashboard (more can be added from there).
+      '';
+      example = ''
+        [
+          "admin:password"
+        ]
+      '';
+      type = types.listOf types.str;
+    };
+
+    openFirewall = mkOption {
+      description = ''
+        Whether to open the firewall for Reposilite.
+        This adds `services.reposilite.settings.port` and `services.reposilite.settings.sslPort` (if SSL is enabled)  to `networking.firewall.allowedTCPPorts`.
+      '';
+      default = false;
+      type = types.bool;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.reposilite = {
+      description = "Reposilite server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        ExecStart =
+          let
+            config-cdn = settingsFormat.generate "reposilite.cdn" cfg.settings;
+            tokens = lib.concatStringsSep " " (map (t: "--token ${t}") cfg.tokens);
+          in
+          "${getExe cfg.package} --local-configuration ${config-cdn} --local-configuration-mode none --working-directory /var/lib/reposilite ${tokens}";
```

Comment:
I'm not familiar with this service, but are there any better ways to provide the tokens to the program than `--token`? Cli args are available to all users on a system via `/proc/<pid>/cmdline`. Environment vars or files would be much more preferrable

Reply:
Generally, you tend to provide a token via the `--token` flag once on the first run, and from then on you generate them in the web ui. 

I'm not even sure managing them through Nix makes sense, thinking about it. Removing a token from the Nix options wouldn't delete it. Need to think about the best way to have the user provide an initial token...